api/pkg/apis/v1alpha1/managers/targets/targets-manager.go

```diff
@@ -112,9 +112,15 @@ func (t *TargetsManager) ReportState(ctx context.Context, current model.TargetSt
 		return model.TargetState{}, err
 	}
 
-	dict := target.Body.(map[string]interface{})
+	dict := map[string]interface{}{}
+	if target.Body != nil {
+		dict = target.Body.(map[string]interface{})
+	}
 
-	specCol := dict["spec"].(model.TargetSpec)
+	specCol := model.TargetSpec{}
+	if (dict["spec"] != nil) && (dict["spec"] != "") {
+		specCol = dict["spec"].(model.TargetSpec)
```

Comment:
Hi, @Haishi2016 . It's a very good point to check client input.
After a second look, the target.Body and dict["spec"] in this function which have type cast are both from state provider. I think the values stored in state provider should be aligned with type definition and may not need to check if type cast is possible. Do you have different opinion?

Reply:
This is called from the vendor, which doesn't do the check, correct? There's a historical reason of using map[string]inteface{} initially, because a partner we worked with POSTed their own status format instead of ours. They embedded Symphony status in their payloads. We can ignore that scenario for now but I think the check is still necessary.

Reply:
We don't do type cast on input variables from vendors. Instead, it's on the state we get from state provider.
The current logic expects k8s or memory states stores status in a structure like "target.spec.status". 
If we want to store target state in a new database, it needs to follow the same protocol.